doc/source/theory/fluid_dynamics/linear_solvers.rst

```diff
@@ -1,4 +1,154 @@
+
+* In Lethe we use the Trilinos implementation of the ILU method through `deal.II <https://www.dealii.org/>`_: `TrilinosWrappers::PreconditionILU <https://dealii.org/developer/doxygen/deal.II/classTrilinosWrappers_1_1PreconditionILU.html>`_. This is of type *ILU(k)* (where ILU(0) is also included). To use this preconditioner, one must specify the absolute and relative tolerance, and the fill level. If the iterative method fails, the fill level is increased by 1 and the solution is attempted again.
+
+* For detailed information on the theory, algorithms and implementation of the different ILU versions, we recommend the book `Iterative Methods for Sparse Linear Systems <https://books.google.ca/books/about/Iterative_Methods_for_Sparse_Linear_Syst.html?id=qtzmkzzqFmcC&redir_esc=y>`_ by Y. Saad.
+
+Algebraic multigrid (AMG)
+__________________________
+
+The main idea of multigrid methods is to efficiently correct all components of the error of the solution on the fine level by using a hierarchy of levels composed by coarser grids. In the algebraic multigrid method, the levels are generated based on the algebraic structure of the system matrix :math:`\mathcal{A}`. It requires the definition of three key components:
+
+* Smoother: it is in charge of smoothing the solution of the residual equations at each level. In general, one uses stationary iterative methods, such as Jacobi or Gauss-Seidel or incomplete LU. The latter two are known for having difficulties as preconditioners for saddle-point problems, therefore, in Lethe we use ILU(k). 
+
+* Intergrid operators: operators that allow to move between the different levels. They are also known as prolongation (:math:`P`) and restriction (:math:`R`) and are used to define the matrices of each level :math:`l`: :math:`\mathcal{A}^{l-1} = R \mathcal{A}^l P`.
+
+* Coarse grid solver: performs a correction of the solution in the coarser level of the hierarchy. In Lethe we use also an ILU decomposition.
+
+Some remarks:
+
+* AMG is more expensive than ILU in terms of setup time and cost per iteration, but it accelerates convergence more significantly.
+
+* There are two types of AMG methods: the classical AMG and smoothed aggregation AMG, which differ in their coarsening strategy that in turn leads to differences also in the prolongation and restriction operators. 
+
+* In Lethe we use the Trilinos implementation of the AMG method through `deal.II <https://www.dealii.org/>`_: `TrilinosWrappers::PreconditionAMG <https://dealii.org/developer/doxygen/deal.II/classTrilinosWrappers_1_1PreconditionAMG.html>`_. One must specify several parameters related to the number of cycles, the type of cycle and smoother parameters. A good starting point to understand all the details of AMG is given in `A Multigrid Tutorial <https://www.researchgate.net/publication/220690328_A_Multigrid_Tutorial_2nd_Edition>`_.
```

Comment:
The reference to A Multigrid Tutorial could be presented in the first paragraph of this subsection. Like this, the reader can refer to it will reading the theory guide :) Same for the ILU and the book Iterative Methods for Sparse Linear Systems.